homework4.txt

```diff
@@ -0,0 +1,261 @@
+
+var3:
+[dmitrii@localhost ~]$ ps
+  PID TTY          TIME CMD
+ 7567 pts/0    00:00:00 bash
+ 7718 pts/0    00:00:00 sleep
+ 7719 pts/0    00:00:00 sleep
+ 7720 pts/0    00:00:00 sleep
+ 7747 pts/0    00:00:00 ps
+
+[dmitrii@localhost ~]$ kill -19 7719 7718 7720
+[dmitrii@localhost ~]$ jobs
+[1]   Stopped                 sleep 100000
+[2]-  Stopped                 sleep 200000
+[3]+  Stopped                 sleep 300000
+
+3) Terminate one of them. (Any)
+[dmitrii@localhost ~]$ kill -15 %1
+
+[1]   Stopped                 sleep 100000
+[dmitrii@localhost ~]$ jobs
+[1]   Завершено      sleep 100000
+[2]-  Stopped                 sleep 200000
+[3]+  Stopped                 sleep 300000
+
+4) To other send a SIGCONT in two different ways.
+[dmitrii@localhost ~]$ kill -18 %2 %3
+[dmitrii@localhost ~]$ jobs
+[2]-  Running 		      sleep 200000 &
+[3]+  Running                 sleep 300000 &
+
+[dmitrii@localhost ~]$ fg %3
+sleep 300000
+
+5) Kill one by PID and the second one by job ID
+var1:
+[dmitrii@localhost ~]$ kill -15 %1 %2
+
+[1]-  Stopped                 sleep 200000
+
+[2]+  Stopped                 sleep 300000
+
+var2:
+[dmitrii@localhost ~]$ kill -9 7804 7805
+[dmitrii@localhost ~]$ jobs
+[3]-  Убито              sleep 200000
+[4]+  Убито              sleep 300000
+
+1.2 Learn lsof command:
+1) Run a "sleep" command, redirect stdout and stderr into two different files.
+[dmitrii@localhost ~]$ touch output.txt; touch err.txt; sleep 400000 >> output.txt 2>>err.txt
+
+2) Относительно lsof справка man сообщает, что lsof есть сокращение от LiSt of Open Files, и что утилита эта служит для вывода информации о том, какие файлы используются теми или иными процессами.
+
+[root@localhost dmitrii]# yum install lsof
+[dmitrii@localhost ~]$ sleep 400000 >> output.txt 2>>err.txt
+^Z
+[1]+  Stopped                 sleep 400000 >> output.txt 2>> err.txt
+[dmitrii@localhost ~]$ ps
+  PID TTY          TIME CMD
+ 7567 pts/0    00:00:00 bash
+11744 pts/0    00:00:00 sleep
+11745 pts/0    00:00:00 ps
+[dmitrii@localhost ~]$ lsof -p 11744
+COMMAND   PID    USER   FD   TYPE DEVICE  SIZE/OFF     NODE NAME
+sleep   11744 dmitrii  cwd    DIR  253,0      4096  4200722 /home/dmitrii
+sleep   11744 dmitrii  rtd    DIR  253,0       238       64 /
+sleep   11744 dmitrii  txt    REG  253,0     33120 12603540 /usr/bin/sleep
+sleep   11744 dmitrii  mem    REG  253,0 106070960 12811050 /usr/lib/locale/locale-archive
+sleep   11744 dmitrii  mem    REG  253,0   2173512    19218 /usr/lib64/libc-2.17.so
+sleep   11744 dmitrii  mem    REG  253,0    164240    19211 /usr/lib64/ld-2.17.so
+sleep   11744 dmitrii    0u   CHR  136,0       0t0        3 /dev/pts/0
+sleep   11744 dmitrii    1w   REG  253,0         0  4831429 /home/dmitrii/output.txt
+sleep   11744 dmitrii    2w   REG  253,0         0  4831430 /home/dmitrii/err.txt 
+
+3)List all ESTABLISHED TCP connections with lsof (only them, without any other commands in a pipe)
+
+MacBook-Air-User:~ user$ lsof -i 6
```

Comment:
I meant not IPv6 but IPv4, and also there is no filter to print only established connections